subrootpaths.go

```diff
@@ -133,8 +127,15 @@ func GetSubrootPaths(squareSize uint, idxStart uint, shareCount uint) ([][][]int
 		return nil, srpInvalidShareCount
 	}
 
+	isn := idxStart + shareCount
+	if isn < idxStart || isn < shareCount {
+		return nil, errors.New("invalid idxStart+shareCount value")
```

Comment:
The max share count possible in Celestia is bound by the maxchunks of the erasure coding library. So this check, while cool, is not a code path that will ever be hit in practice. If you look at nmt independently as a library it makes more sense though.

Reply:
Much safer to not bake those assumptions into a library that'll be open to everyone to use. I was able to hit that point from trying to attack this library. However I do understand the context.